blueflood-core/src/test/java/com/rackspacecloud/blueflood/io/serializers/StringSerializerTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2016 Rackspace
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.rackspacecloud.blueflood.io.serializers;
+
+import com.netflix.astyanax.model.ColumnFamily;
+import com.rackspacecloud.blueflood.exceptions.SerializationException;
+import com.rackspacecloud.blueflood.exceptions.UnexpectedStringSerializationException;
+import com.rackspacecloud.blueflood.io.Constants;
+import com.rackspacecloud.blueflood.types.Locator;
+import com.rackspacecloud.blueflood.types.SimpleNumber;
+import org.apache.commons.codec.binary.Base64;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.*;
+import java.nio.ByteBuffer;
+
+public class StringSerializerTest {
+
+    private String TEST_STRING = "This is a test string.";
+
+    @Test(expected = SerializationException.class)
+    public void testSerializerForStringShouldFail() throws Throwable {
+        try {
+            NumericSerializer.serializerFor(String.class);
+        } catch (RuntimeException re) {
+            throw re.getCause();
+        }
+    }
+
+    // we used to allow de-serializing strings, but we don't anymore.
+    // catch that error and assert it happens only when expected.
+    @Test (expected = UnexpectedStringSerializationException.class)
+    public void testStringFullSerializationShouldFail() throws Throwable {
+        ByteArrayOutputStream baos = serializeString(TEST_STRING);
+        ByteBuffer byteBuffer = ByteBuffer.wrap(baos.toByteArray());
+        try {
+            NumericSerializer.serializerFor(Object.class).fromByteBuffer(byteBuffer);
+        } catch (RuntimeException re) {
+            throw re.getCause();
+        }
```

Comment:
The NumericSerializer.serializerFor() code wraps any UnexpectedStringSerializerException into a RuntimeException. I want to check specificly for the UnexpectedStringSerializerException, so if I'm getting a different exception, it should be considered test failure.